lib/common/mem.h

```diff
@@ -18,6 +18,7 @@ extern "C" {
 /*-****************************************
 *  Dependencies
 ******************************************/
+#define ZSTD_DEPS_NEED_SWAP
 #include "zstd_deps.h"  /* size_t, ptrdiff_t, ZSTD_memcpy */
```

Comment:
I understand the short term requirement to get this working fine for kernel and 1.4.6,

but for the sake of code clarity beyond that goal,
I admit that after all these successive changes,
it's no longer clear to me what are the intended scope of `mem.h` and `zstd_deps.h`.

Maybe the purpose of each file is worth documenting in the header of each file,
as there is a risk for them becoming some kind of "bag of everything" after multiple edition rounds.

Reply:
We could move the int types, the swaps, and maybe mem{cpy,move,set} into `mem.h`. Then move some of the extranuous stuff, like the asan/msan poisoning, out of `mem.h`. And have the kernel provide both `zstd_deps.h` and `mem.h`.

Reply:
Yes,
I like the idea of each file having a clean perimeter,
and this proposal makes sense.